packages/dd-trace/src/appsec/iast/analyzers/injection-analyzer.js

```diff
@@ -1,19 +1,25 @@
 'use strict'
 const Analyzer = require('./vulnerability-analyzer')
 const { isTainted, getRanges } = require('../taint-tracking/operations')
+const { SQL_ROW_VALUE } = require('../taint-tracking/source-types')
 
 class InjectionAnalyzer extends Analyzer {
   _isVulnerable (value, iastContext) {
-    if (value) {
-      return isTainted(iastContext, value)
+    if (value && isTainted(iastContext, value)) {
+      return this._areRangesVulnerable(getRanges(iastContext, value))
     }
```

Comment:
here there are two calls to the native part, could be something like that?

```suggestion
    const ranges = value && getRanges(iastContext, value)  // `getRanges(iastContext, not_tainted_value)` returns undefined or an empty array?

    if (ranges) {
      return this._areRangesVulnerable(ranges)
    }
```

Reply:
Yes, I did in this way to prevent creating arrays when it is not tainted,  but if it is not tainted, will create only an empty array, I think that it worth to change to your suggestion. Changing it.